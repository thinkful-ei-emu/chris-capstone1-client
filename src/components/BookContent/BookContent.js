import React from 'react';

export default function BookContent({ book }) {
    return (
        <>
            <p>source: {book.book_source}</p>
            <p>list: {book.list}</p>
            {book.genre && <p>genre: {book.genre}</p>}
            {book.book_report && <p>book report: {book.book_report}</p>}
            {book.rating && <p>your rating: {book.rating}%</p>}
            <p>average user rating: <span>{book.average_rating}%</span></p>
            <p># of ratings: <span>{book.number_of_ratings}</span></p>
        </>
    )
}