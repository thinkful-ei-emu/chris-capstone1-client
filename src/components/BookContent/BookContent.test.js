import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BookContent from './BookContent';
import ReactDom from 'react-dom';

describe('BookContent component', () => {
  let book = {
    id: 1,
    title: 'Disappearing Earth',
    author: 'Julia Phillips',
    list: 'wishlist',
    book_source: 'Print',
    genre: 'Fantasy',
    book_report: 'Nothing to see here',
    image: 'http://placehold.it/500x500',
    rating: 13, 
    recommended: false, 
    poster_rating: null, 
    poster_report: null, 
    poster_id: null,
    user_id: 1,
    date_created: '2029-01-22T16:28:32.615Z'
  }
    it('renders without crashing', () => {
        
        const div = document.createElement('div')
        ReactDom.render(
            <BookContent key={book.id} book={book} />,
          div
        )
        ReactDom.unmountComponentAtNode(div)
      })
    it('renderes BookContent by default', () => {
        expect(toJson(shallow(<BookContent key={book.id} book={book} />))).toMatchSnapshot() 
    })
})