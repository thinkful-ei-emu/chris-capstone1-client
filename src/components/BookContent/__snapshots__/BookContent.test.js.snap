// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookContent component renderes BookContent by default 1`] = `
<Fragment>
  <p>
    source: 
    Print
  </p>
  <p>
    list: 
    wishlist
  </p>
  <p>
    genre: 
    Fantasy
  </p>
  <p>
    book report: 
    Nothing to see here
  </p>
  <p>
    your rating: 
    13
    %
  </p>
  <p>
    average user rating: 
    <span>
      %
    </span>
  </p>
  <p>
    # of ratings: 
    <span />
  </p>
</Fragment>
`;
